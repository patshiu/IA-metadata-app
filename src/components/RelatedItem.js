import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

import './RelatedItem.scss';
import ItemTitle from './ItemTitle';

const RelatedItem = ({ itemSource, itemIdentifier }) => {
  return (
    <div className="related-item-card-container col-1-3 padded">
      <div className="related-item-card">
        <img
          src={'https://archive.org/services/img/' + itemIdentifier}
          className="card-hero"
          alt-text={'thumbnail for ' + ItemTitle}
        />
        <div className="card-content">
          <h3>
            <Link to={'/' + itemIdentifier}>{itemSource.title}</Link>
          </h3>
          <p>Media type: {itemSource.mediatype}</p>
          <p>{itemSource.description ? itemSource.description : ''}</p>
        </div>
      </div>
    </div>
  );
};

RelatedItem.propTypes = {
  itemSource: PropTypes.shape({
    title: PropTypes.oneOfType([PropTypes.string, PropTypes.array]),
    mediatype: PropTypes.oneOfType([PropTypes.string, PropTypes.array]),
    description: PropTypes.oneOfType([PropTypes.string, PropTypes.array])
  }),
  itemIdentifier: PropTypes.string.isRequired
};

export default RelatedItem;
