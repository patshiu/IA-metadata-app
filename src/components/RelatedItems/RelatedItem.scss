@import '../../_variables';

.related-item-card-container {
  padding: 0.25rem;

  .related-item-card {
      border: 1px solid $base-darker;
      border-radius: $border-radius-md;
      overflow: hidden;
      background-color: $base-lighter;
      box-sizing: border-box;
      padding: 0 0 2rem 0;
      margin: 0.5rem;

      .card-hero {
        width: 100%;
        height: auto;
        padding: 0;
        margin: 0;
      }

      .card-content {
        box-sizing: border-box;
        padding: 0 1.2rem 0 1rem;
      }
    }

}
