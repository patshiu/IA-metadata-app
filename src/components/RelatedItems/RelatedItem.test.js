import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';

import RelatedItem from './RelatedItem';

it('renders related item card correctly', () => {
  const itemSource = {
    creatorSorter: ['levene (carol)'],
    description: [
      'Depicts actual construction progress throughout the San Francisco Bay Area Rapid Transit System during 1967-68, emphasizing the original and continuing aims of the district--to provide safe, fast, economical and comfortable mass transit, thereby eliminating traffic congestion and preserving the beauty and character of the area.'
    ],
    title: ['Along the Way'],
    downloads: [7755],
    mediatype: ['movies'],
    collection: [
      'prelinger',
      'fav-david_silva',
      'fav-davykurlander',
      'fav-thomas_lopez794',
      'fav-breudd',
      'fav-meateye',
      'fav-smardzz',
      'fav-fruitiebat',
      'fav-project_greenpark',
      'fav-foolishblue',
      'fav-jordi_mumma',
      'fav-xeconya',
      'fav-thevelvetundergrad',
      'fav-thegreenflyingkiwi',
      'fav-chriscousin',
      'fav-margaret_b_1',
      'fav-iowatvman',
      'fav-blake_rand'
    ],
    num_reviews: [4],
    subjectSorter: ['need keyword'],
    publicdate: ['2003-05-16T10:39:21Z']
  };
  const itemIdentifier = 'along_the_way';
  const tree = renderer
    .create(
      <Router>
        <RelatedItem itemSource={itemSource} itemIdentifier={itemIdentifier} />
      </Router>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
