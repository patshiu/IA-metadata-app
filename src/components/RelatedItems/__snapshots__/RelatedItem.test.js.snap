// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders related item card correctly 1`] = `
<div
  className="related-item-card-container col-1-3 padded"
>
  <div
    className="related-item-card"
  >
    <img
      alt="Thumbnail for Along the Way"
      className="card-hero"
      src="https://archive.org/services/img/along_the_way"
    />
    <div
      className="card-content"
    >
      <h3>
        <strong>
          <a
            href="/along_the_way"
            onClick={[Function]}
          >
            Along the Way
          </a>
        </strong>
      </h3>
      <p>
        Media type: 
        movies
      </p>
      <p>
        Depicts actual construction progress throughout the San Francisco Bay Area Rapid Transit System during 1967-68, emphasizing the original and continuing aims of the district--to provide safe, fast, economical and comfortable mass transit, thereby eliminating traffic congestion and preserving the beauty and character of the area.
      </p>
    </div>
  </div>
</div>
`;
